src/main/java/com/spotify/reaper/storage/CassandraStorage.java

```diff
@@ -142,7 +153,7 @@ public boolean isStorageConnected() {
   @Override
   public boolean addCluster(Cluster cluster) {
     try {
-      session.execute(insertClusterPrepStmt.bind(cluster.getName(), cluster.getPartitioner(), cluster.getSeedHosts()));
+      session.executeAsync(insertClusterPrepStmt.bind(cluster.getName(), cluster.getPartitioner(), cluster.getSeedHosts()));
```

Comment:
It was on purpose.

> This will lead to potential silent failures and chained calls being inconsistent (we may try to read something that isn't yet written to all replicas).

Can you describe any situations where and how this would happen?

I don't know the code thoroughly, but from what i've seen so far:
 - within this request the `cluster` object exists in-memory, and is used as such,
 - whether a separate request doesn't yet find it is part of the already existing eventual consistency of the design.